### PR TITLE
Allow Agora to listen to multiple interfaces

### PR DESCRIPTION
**Pull request by Geod24**
Contains a bunch of refactoring, but the end goal is to make Agora able to listen either on HTTP or RPC.
Part of #203 .

### CODE REVIEW

**Comment by Geod24** [comment]
Don't merge, I will probably add a second interface in the system integration test to make sure this works properly.
So Faucet and the test would contact Agoras on HTTP, and the Agoras would talk to themselves via another interface.
I might even mix the interfaces just so that we get a very good coverage.

**Comment by Geod24** [comment]
Damn, found a Formatter bug: https://github.com/sociomantic-tsunami/ocean/issues/846

**Comment by Geod24** [comment]
Can I haz review ?

File: source/agora/common/Config.d
Comment (linked0) [inline source/agora/common/Config.d:132]: Kind of a strange format for the comment.

File: source/agora/common/Config.d
Comment (linked0) [inline source/agora/common/Config.d]: There are unnecessary spaces and a period.

File: source/agora/common/Config.d
Comment (linked0) [inline source/agora/common/Config.d:436]: Why don't you make a function for that, like `parseInterfaceConfig`?

File: source/agora/node/Runner.d
Comment (linked0) [inline source/agora/node/Runner.d:130]: `interf` would be better than `interface_`.

**Review by linked0 — APPROVED** [review]
LGTM just some nits

File: source/agora/node/Runner.d
Comment (Geod24) [reply source/agora/node/Runner.d:130]: https://dlang.org/dstyle.html#naming_keywords

File: source/agora/common/Config.d
Comment (Geod24) [reply source/agora/common/Config.d]: Ah, it'll align in a future commit, with the TCP interface

File: source/agora/common/Config.d
Comment (Geod24) [reply source/agora/common/Config.d:132]: It's because the two lines are commented out. If I used `///` the documentation would incorrectly be applied to the next member. It isn't great but I hope to get this bug fixed in the near-ish future.

File: source/agora/common/Config.d
Comment (Geod24) [reply source/agora/common/Config.d]: But I removed them in the meantime

File: source/agora/common/Config.d
Comment (Geod24) [reply source/agora/common/Config.d:436]: Feels a bit overkill, given the simplicity of this code. Also, `parseSequence` should really be able to handle this.

**Comment by Geod24** [comment]
Rebased and fixed spaces, as it was green before, merging.

**Pull request merged**
